From 8fcf6db6d06c3eb950d62c440786d987558fd2b4 Mon Sep 17 00:00:00 2001
From: Alexander Rukletsov <alexr@apache.org>
Date: Mon, 13 Aug 2018 20:05:26 +0200
Subject: [PATCH 04/16] Added 'received' timestamp into `process::Request`.

This allows us to note when a request comes in and
hence calculate request processing time.

Review: https://reviews.apache.org/r/68992
(cherry picked from commit 6ab80da5b2770a96802f7893ac092d6c8a92a824)
---
 3rdparty/libprocess/include/process/http.hpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/3rdparty/libprocess/include/process/http.hpp b/3rdparty/libprocess/include/process/http.hpp
index fa66b0fe2..a1b0bb9e3 100644
--- a/3rdparty/libprocess/include/process/http.hpp
+++ b/3rdparty/libprocess/include/process/http.hpp
@@ -27,6 +27,7 @@
 #include <boost/functional/hash.hpp>
 
 #include <process/address.hpp>
+#include <process/clock.hpp>
 #include <process/future.hpp>
 #include <process/owned.hpp>
 #include <process/pid.hpp>
@@ -517,7 +518,7 @@ public:
 struct Request
 {
   Request()
-    : keepAlive(false), type(BODY) {}
+    : keepAlive(false), type(BODY), received(Clock::now()) {}
 
   std::string method;
 
@@ -562,6 +563,8 @@ struct Request
   std::string body;
   Option<Pipe::Reader> reader;
 
+  Time received;
+
   /**
    * Returns whether the encoding is considered acceptable in the
    * response. See RFC 2616 section 14.3 for details.
-- 
2.21.0

